src/main/scala/chisel/lib/dclib/DCOperator.scala

```diff
@@ -0,0 +1,42 @@
+package chisel.lib.dclib
+
+import chisel3._
+import chisel3.util._
+
+/**
+ * This module declares a multi-input decoupled operator.  This is an example using showing how to use
+ * DCInput and DCOutput modules to create a module with registered-output timing.
+ *
+ * @param n      The number of inputs for the operator
+ * @param width  The bit-width of inputs and outputs
+ * @param op     Function with the required operator
+ */
+class DCOperator(n: Int, width: Int, op: (UInt, UInt) => UInt) extends Module {
+  val io = IO(new Bundle {
+    val a = Vec(n, Flipped(Decoupled(UInt(width.W))))
+    val z = Decoupled(UInt(width.W))
+  })
+  require (n >= 2)
+  val a_int = for (n <- 0 until n) yield DCInput(io.a(n))
+  val z_dcout = Module(new DCOutput(UInt(width.W)))
```

Comment:
Nit: I'd use the companion object here since the PR uses it for `DCInput`:
```suggestion
  val z_dcout = DCOutput(UInt(width.W))
```